Projects/UOContent/Gumps/Base/GumpSystem.cs

```diff
@@ -161,12 +134,19 @@ private static void Send(NetState ns, BaseGump gump, bool singleton)
 
             if (!replaced)
             {
+                if (list.Count >= GumpCap)
+                {
+                    _logger.Information("Exceeded gump cap, disconnecting...");
+                    ns.Disconnect("Exceeded gump cap.");
```

Comment:
Interesting.  I don't have all the context, but I wonder if it would be better to start recycling the gumps on the NetState rather than disconnecting it.  I could foresee a situation where this happens due to something the player user has no control over.  That sounds frustrating.